sphinx/domains/rst.py

```diff
@@ -278,8 +279,9 @@ def resolve_any_xref(self, env: BuildEnvironment, fromdocname: str, builder: Bui
                          ) -> list[tuple[str, Element]]:
         results: list[tuple[str, Element]] = []
         for objtype in self.object_types:
-            todocname, node_id = self.objects.get((objtype, target), (None, None))
-            if todocname:
+            result = self.objects.get((objtype, target))
+            if result:
+                todocname, node_id = result
```

Comment:
can we keep the tuple unpacking here?

Reply:
No, not really. If you unpack the tuple before checking the `get` call returned anything, there's no way to know (from a type safety perspective) that just because `todocname` is set, `node_id` must also be set. By promoting the 'Optional' to the level of the tuple, and moving the tuple unpacking inside the if statement, then everything is correctly typed

Reply:
I *could* use the walrus operator here to shave off a line. Would still need to unpack the tuple inside the if block